src/reports.cpp

```diff
@@ -944,8 +953,12 @@ static int attack_info(const reports::context& rc, const attack_type &at, config
 		const std::string spacer = "misc/blank.png~CROP(0, 0, 16, 21)"; // 21 == 16+5
 		add_image(res, spacer + "~BLIT(" + range_png + ",0,5)", damage_versus.tooltip);
 		add_image(res, spacer + "~BLIT(" + type_png + ",0,5)", damage_versus.tooltip);
-		if(secondary_type_png_exists){
-			add_image(res, spacer + "~BLIT(" + secondary_type_png + ",0,5)", damage_versus.tooltip);
+		if(!secondary_types_png.empty()){
```

Comment:
If the vector is empty the loop will do nothing anyway.

Reply:
if(!secondary_types_png.empty()){ removed